From 3679ca623a641886119d1427f5bb8fc093e3474b Mon Sep 17 00:00:00 2001
From: unknown author <cooker@mandrivalinux.org>
Date: Tue, 10 Feb 2009 15:47:49 +0000
Subject: gcc43 hash style gnu

2006-07-11  Jakub Jelinek  <jakub@redhat.com>

        * config/i386/linux.h (LINK_SPEC): Add --hash-style=gnu.
        * config/i386/linux64.h (LINK_SPEC): Likewise.
        * config/rs6000/sysv4.h (LINK_OS_LINUX_SPEC): Likewise.
        * config/rs6000/linux64.h (LINK_OS_LINUX_SPEC32,
        LINK_OS_LINUX_SPEC64): Likewise.
        * config/s390/linux.h (LINK_SPEC): Likewise.
        * config/ia64/linux.h (LINK_SPEC): Likewise.
        * config/sparc/linux.h (LINK_SPEC): Likewise.
        * config/sparc/linux64.h (LINK_SPEC, LINK_ARCH32_SPEC,
        LINK_ARCH64_SPEC): Likewise.
        * config/alpha/linux-elf.h (LINK_SPEC): Likewise.

Rebased for 4.7.1
[changes to i386/linux(64).h are now in i386/gnu-user(64).h / tmb]
Signed-off-by: Thomas Backlund <tmb@mageia.org>

---
 gcc/config/alpha/linux-elf.h |    2 +-
 gcc/config/i386/gnu-user.h   |    2 +-
 gcc/config/i386/gnu-user64.h |    2 +-
 gcc/config/ia64/linux.h      |    2 +-
 gcc/config/rs6000/linux64.h  |    4 ++--
 gcc/config/rs6000/sysv4.h    |    2 +-
 gcc/config/s390/linux.h      |    2 +-
 gcc/config/sparc/linux.h     |    2 +-
 gcc/config/sparc/linux64.h   |    6 +++---
 9 files changed, 12 insertions(+), 12 deletions(-)

diff -Nurp gcc-4.7.1/gcc/config/alpha/linux-elf.h gcc-4.7.1-hash-style-gnu/gcc/config/alpha/linux-elf.h
--- gcc-4.7.1/gcc/config/alpha/linux-elf.h      2011-04-11 21:46:05.000000000 +0300
+++ gcc-4.7.1-hash-style-gnu/gcc/config/alpha/linux-elf.h       2012-06-24 01:05:44.170141253 +0300
@@ -38,7 +38,7 @@ along with GCC; see the file COPYING3.
 
 #define ELF_DYNAMIC_LINKER     GNU_USER_DYNAMIC_LINKER
 
-#define LINK_SPEC "-m elf64alpha %{G*} %{relax:-relax}         \
+#define LINK_SPEC "-m elf64alpha --hash-style=gnu %{G*} %{relax:-relax}        \
   %{O*:-O3} %{!O*:-O1}                                         \
   %{shared:-shared}                                            \
   %{!shared:                                                   \
diff -Nurp gcc-4.7.1/gcc/config/i386/gnu-user64.h gcc-4.7.1-hash-style-gnu/gcc/config/i386/gnu-user64.h
--- gcc-4.7.1/gcc/config/i386/gnu-user64.h      2011-07-07 18:38:34.000000000 +0300
+++ gcc-4.7.1-hash-style-gnu/gcc/config/i386/gnu-user64.h       2012-06-24 02:17:03.415074353 +0300
@@ -76,7 +76,7 @@ see the files COPYING3 and COPYING.RUNTI
 #define LINK_SPEC "%{" SPEC_64 ":-m " GNU_USER_LINK_EMULATION64 "} \
                    %{" SPEC_32 ":-m " GNU_USER_LINK_EMULATION32 "} \
                    %{" SPEC_X32 ":-m " GNU_USER_LINK_EMULATIONX32 "} \
-  %{shared:-shared} \
+   --hash-style=gnu %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
diff -Nurp gcc-4.7.1/gcc/config/i386/gnu-user.h gcc-4.7.1-hash-style-gnu/gcc/config/i386/gnu-user.h
--- gcc-4.7.1/gcc/config/i386/gnu-user.h        2011-05-05 15:32:50.000000000 +0300
+++ gcc-4.7.1-hash-style-gnu/gcc/config/i386/gnu-user.h 2012-06-24 02:15:12.360158410 +0300
@@ -98,7 +98,7 @@ along with GCC; see the file COPYING3.
   { "dynamic_linker", GNU_USER_DYNAMIC_LINKER }
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m %(link_emulation) %{shared:-shared} \
+#define LINK_SPEC "-m %(link_emulation) --hash-style=gnu %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
diff -Nurp gcc-4.7.1/gcc/config/ia64/linux.h gcc-4.7.1-hash-style-gnu/gcc/config/ia64/linux.h
--- gcc-4.7.1/gcc/config/ia64/linux.h   2011-06-03 21:30:39.000000000 +0300
+++ gcc-4.7.1-hash-style-gnu/gcc/config/ia64/linux.h    2012-06-24 01:05:44.171141284 +0300
@@ -59,7 +59,7 @@ do {                                          \
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
-#define LINK_SPEC "\
+#define LINK_SPEC "--hash-style=gnu \
   %{shared:-shared} \
   %{!shared: \
     %{!static: \
diff -Nurp gcc-4.7.1/gcc/config/rs6000/linux64.h gcc-4.7.1-hash-style-gnu/gcc/config/rs6000/linux64.h
--- gcc-4.7.1/gcc/config/rs6000/linux64.h       2012-03-12 18:16:51.000000000 +0200
+++ gcc-4.7.1-hash-style-gnu/gcc/config/rs6000/linux64.h        2012-06-24 01:08:29.690420564 +0300
@@ -375,11 +375,11 @@ extern int dot_symbols;
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER64, UCLIBC_DYNAMIC_LINKER64)
 
 
-#define LINK_OS_LINUX_SPEC32 "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC32 "-m elf32ppclinux --hash-style=gnu %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER32 "}}"
 
-#define LINK_OS_LINUX_SPEC64 "-m elf64ppc %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC64 "-m elf64ppc --hash-style=gnu %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER64 "}}"
 
diff -Nurp gcc-4.7.1/gcc/config/rs6000/sysv4.h gcc-4.7.1-hash-style-gnu/gcc/config/rs6000/sysv4.h
--- gcc-4.7.1/gcc/config/rs6000/sysv4.h 2012-04-30 20:39:01.000000000 +0300
+++ gcc-4.7.1-hash-style-gnu/gcc/config/rs6000/sysv4.h  2012-06-24 01:05:44.172141315 +0300
@@ -814,7 +814,7 @@ extern int fixuplabelno;
 #define GNU_USER_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 
-#define LINK_OS_LINUX_SPEC "-m elf32ppclinux %{!shared: %{!static: \
+#define LINK_OS_LINUX_SPEC "-m elf32ppclinux --hash-style=gnu %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
   -dynamic-linker " GNU_USER_DYNAMIC_LINKER "}}"
 
diff -Nurp gcc-4.7.1/gcc/config/s390/linux.h gcc-4.7.1-hash-style-gnu/gcc/config/s390/linux.h
--- gcc-4.7.1/gcc/config/s390/linux.h   2011-06-03 21:30:39.000000000 +0300
+++ gcc-4.7.1-hash-style-gnu/gcc/config/s390/linux.h    2012-06-24 01:05:44.173141344 +0300
@@ -66,7 +66,7 @@ along with GCC; see the file COPYING3.
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
-  "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
+  "%{m31:-m elf_s390}%{m64:-m elf64_s390} --hash-style=gnu \
    %{shared:-shared} \
    %{!shared: \
       %{static:-static} \
diff -Nurp gcc-4.7.1/gcc/config/sparc/linux64.h gcc-4.7.1-hash-style-gnu/gcc/config/sparc/linux64.h
--- gcc-4.7.1/gcc/config/sparc/linux64.h        2011-11-12 03:05:07.000000000 +0200
+++ gcc-4.7.1-hash-style-gnu/gcc/config/sparc/linux64.h 2012-06-24 01:05:44.173141344 +0300
@@ -105,7 +105,7 @@ along with GCC; see the file COPYING3.
   { "link_arch_default", LINK_ARCH_DEFAULT_SPEC },       \
   { "link_arch",        LINK_ARCH_SPEC },
 
-#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/usr/lib %{shared:-shared} \
+#define LINK_ARCH32_SPEC "-m elf32_sparc --hash-style=gnu -Y P,%R/usr/lib %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -113,7 +113,7 @@ along with GCC; see the file COPYING3.
       %{static:-static}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc --hash-style=gnu -Y P,%R/usr/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
@@ -202,7 +202,7 @@ extern const char *host_detect_local_cpu
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc --hash-style=gnu -Y P,%R/usr/lib64 %{shared:-shared} \
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
diff -Nurp gcc-4.7.1/gcc/config/sparc/linux.h gcc-4.7.1-hash-style-gnu/gcc/config/sparc/linux.h
--- gcc-4.7.1/gcc/config/sparc/linux.h  2011-11-12 03:05:07.000000000 +0200
+++ gcc-4.7.1-hash-style-gnu/gcc/config/sparc/linux.h   2012-06-24 01:05:44.173141344 +0300
@@ -87,7 +87,7 @@ extern const char *host_detect_local_cpu
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc --hash-style=gnu -Y P,/usr/lib %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!static: \ 
